npm ERR! grunt-casperjs@1.0.10 install: `node install.js`

I get an error when installing via NPM. 

Error is:

/Users/blah/somefolder/node_modules/grunt-casperjs/node_modules/adm-zip/zipFile.js:66
            throw Utils.Errors.INVALID_FORMAT;
                              ^
Invalid or unsupported zip format. No END header found

npm ERR! grunt-casperjs@1.0.10 install: `node install.js`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the grunt-casperjs@1.0.10 install script.
npm ERR! This is most likely a problem with the grunt-casperjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls grunt-casperjs
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 13.1.0
npm ERR! command "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! cwd /Users/blah/somefolder
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/blah/somefolder/npm-debug.log
npm ERR! not ok code 0

--- a/install.js
+++ b/install.js
@@ -8,6 +8,7 @@ var path = require('path')
 var url = require('url')
 var rimraf = require('rimraf').sync
 var AdmZip = require('adm-zip')
+var request = require('request')
 
 fs.existsSync = fs.existsSync || path.existsSync
 
@@ -22,7 +23,7 @@ function isCasperInstalled(notInstalledCallback) {
     // Note that "which" doesn't work on windows.
     cp.exec("casperjs --version", function(error, stdout, stderr) {
         if ( error ) {
-            console.log("Casperjs not installed.  Installing.");
+            console.log("Casperjs not installed.  Installing...");
             notInstalledCallback();
         } else {
             var casperVersion = stdout.replace(/^\s+|\s+$/g,'');
@@ -56,27 +57,40 @@ function unzipTheZippedFile() {
 }
 
 function downloadZipFromGithub() {
-    var file = fs.createWriteStream(path.join(tmpPath, "archive.zip"));
-    var lengthSoFar = 0;
-    var request = https.get(downloadUrl, function(response) {
-        if (response.statusCode === 301 || response.statusCode === 302) {
-            downloadUrl = response.headers.location;
-            downloadZipFromGithub();
+    var requestOptions = {
+        uri: downloadUrl,
+        encoding: null, // Get response as a buffer
+        followRedirect: true, // The default download path redirects to a CDN URL.
+        headers: {}
+    };
+
+    request(requestOptions, function(error, response, body) {
+        if (!error && response.statusCode === 200) {
+            fs.writeFileSync(path.join(tmpPath, "archive.zip"), body);
+
+            console.log( Math.floor(body.length / 1025) + "KB received. Unzipping...");
+            unzipTheZippedFile();
+        } else if (response) {
+            console.error('Error requesting archive.\n' +
+                  'Status: ' + response.statusCode + '\n' +
+                  'Request options: ' + JSON.stringify(requestOptions, null, 2) + '\n' +
+                  'Response headers: ' + JSON.stringify(response.headers, null, 2) + '\n');
+
+            tidyUp();
+            exit(1);
+        } else if (error) {
+            console.error('Error making request.\n' + error.stack + '\n\n' +
+              'Please report this full log at https://github.com/ronaldlokers/grunt-casperjs');
+
+            tidyUp();
+            exit(1);
         } else {
-            response.pipe(file);
-            response.on('data', function(chunk) {
-                console.log('Receiving ' + Math.floor((lengthSoFar += chunk.length) / 1024) + 'K...' );
-            }).
-                on('end', unzipTheZippedFile).
-                on('error', function(e) {
-                    console.log('An error occured whilst trying to download Casper.JS ' + e.message);
-                    tidyUp();
-                });
-        }
-    });
-    request.on('error', function(e) {
-        console.log('An error occured whilst trying to download Casper.JS ' + e.message);
-        tidyUp();
+            console.error('Something unexpected happened, please report this full ' +
+              'log at https://github.com/ronaldlokers/grunt-casperjs');
+
+            tidyUp();
+            exit(1);
+       }
     });
 }
 

--- a/package.json
+++ b/package.json
@@ -8,20 +8,20 @@
     "email": "ronaldlokers@me.com",
     "url": "http://www.ronaldlokers.nl"
   },
- "contributors": [
-   {
-     "name": "Ronald Lokers",
-     "email": "ronaldlokers@me.com",
-     "url": "http://www.ronaldlokers.nl"
-   },
-   {
-     "name": "Nick Sullivan",
-     "email": "nick@sullivanflock.com",
-     "url": "http://creationeer.me"
-   },
-   {
-     "name": "Jatin Chopra",
-     "url": "https://github.com/jatin-"
+  "contributors": [
+    {
+      "name": "Ronald Lokers",
+      "email": "ronaldlokers@me.com",
+      "url": "http://www.ronaldlokers.nl"
+    },
+    {
+      "name": "Nick Sullivan",
+      "email": "nick@sullivanflock.com",
+      "url": "http://creationeer.me"
+    },
+    {
+      "name": "Jatin Chopra",
+      "url": "https://github.com/jatin-"
     }
   ],
   "repository": {
@@ -48,7 +48,8 @@
   "dependencies": {
     "phantomjs": "~1.9.1",
     "rimraf": "~2.1.4",
-    "adm-zip": "~0.4.3"
+    "adm-zip": "~0.4.3",
+    "request": "~2.36.0"
   },
   "devDependencies": {
     "grunt-contrib-jshint": "~0.1.1",

